### PR TITLE
api: validate workflow name in create_workflow_from_json

### PR DESCRIPTION
**Pull request by audrium**
closes https://github.com/reanahub/reana-client/issues/584

To test make sure it works:
- using `reana-client`:
```
$ reana-client create -w test.41
 ==> ERROR: Workflow name test.41 contains illegal character "."
 ```
 - [using Python API](https://blog.reana.io/posts/2021/reana-client-python-api/)

### CODE REVIEW

File: reana_client/utils.py [hunk 1]
```diff
@@ -184,6 +184,18 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def validate_workflow_name(workflow_name: str) -> str:
+    """Validate workflow name."""
+    not_allowed_characters = ["."]
```
Comment (mvidalgarcia) [inline reana_client/utils.py:189]: Minor: Would it make sense to move this to config.py?

File: reana_client/cli/utils.py [hunk 2]
```diff
@@ -237,19 +237,13 @@ def get_formatted_progress(progress):
...
-        for item in not_allowed_characters:
-            if item in workflow_name:
-                display_message(
-                    "Workflow name {} contains illegal "
-                    'character "{}"'.format(workflow_name, item),
-                    msg_type="error",
-                )
-                sys.exit(1)
-    return workflow_name
+def validate_workflow_name_parameter(ctx, _, workflow_name):
```
Comment (mvidalgarcia) [inline reana_client/cli/utils.py:240]: Minor: as we're changing it we can add type hints.

**Review by mvidalgarcia — APPROVED** [review]
Two very minor comments, works as expected 👍 

**PR not checked out:**
```console
$ reana-client list
NAME                                   RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS  
foo.bar.1                                           2022-01-14T16:22:27   -                     -                     created 
```

**PR checked out:**
```python
In [9]: create_workflow_from_json(
   ...:     workflow_json=my_workflow,
   ...:     name=my_workflow_name,
   ...:     access_token=my_reana_token,
   ...:     parameters=my_inputs,
   ...:     workflow_engine=my_workflow_type)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-7d107fc40d7c> in <module>
----> 1 create_workflow_from_json(
      2     workflow_json=my_workflow,
      3     name=my_workflow_name,
      4     access_token=my_reana_token,
      5     parameters=my_inputs,

~/code/reanahub/reana-client/reana_client/api/client.py in create_workflow_from_json(name, access_token, workflow_json, workflow_file, parameters, workflow_engine, outputs)
    226             workflow_engine='serial')
    227     """
--> 228     validate_workflow_name(name)
    229     if is_uuid_v4(name):
    230         raise ValueError("Workflow name cannot be a valid UUIDv4")

~/code/reanahub/reana-client/reana_client/utils.py in validate_workflow_name(workflow_name)
    191         for item in not_allowed_characters:
    192             if item in workflow_name:
--> 193                 raise ValueError(
    194                     f'Workflow name {workflow_name} contains illegal character "{item}"'
    195                 )

ValueError: Workflow name foo.bar contains illegal character "."
```

**Comment by audrium** [comment]
Moved `validate_workflow_name` function to `reana-commons` and reused it inside `reana-server create_workflow` endpoint to validate the workflow name there as well. @mvidalgarcia, please have another look.

**Pull request merged**